src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2044,7 +2053,44 @@ static unsigned int processSubscriptions
       }
     }
 
-    /* Check 2: String Filters */
+    #ifdef EXPR_BASIC
+        bool basic = true;
+    #else
+        bool basic = false;
+    #endif
+
+    // Build JEXL evaluation contexts using entity attributes and their metadata
```

Comment:
I'd suggest to do the JEXL evaluation after string filters (which is less expensive) and before than geo-evaluation (which is more expensive, as it hits de MongoDB).